pkg/machine/e2e/machine_test.go

```diff
@@ -71,10 +71,10 @@ var _ = BeforeSuite(func() {
 		Fail(fmt.Sprintf("failed to pull wsl disk: %q", pullError))
 	}
 	if testProvider.VMType() == define.AppleHvVirt {
-		cmd := exec.Command("softwareupdate", "--install-rosetta", "--agree-to-license")
+		cmd := exec.Command("arch", "-arch", "x86_64", "/usr/bin/uname", "-m")
 		err := cmd.Run()
 		if err != nil {
-			Fail(fmt.Sprintf("Command failed with error: %q", err))
+			Fail(fmt.Sprintf("Rosetta doesn't appear to be installed or functional, 'uname' test command failed with error: %q", err))
 		}
```

Comment:
I think we should move this into the cirrus setup scripts and not the test suite itself

Reply:
I'm concerned if somebody ever wants to run the tests locally, outside of CI, then they'll simply fail.  These failures may be very obscure (I have yet to test this) as the actual binary could be executing inside a VM or inside a container inside a VM (I'm really not sure how these tests work).

Reply:
I'm not opposed to adding the check in both places, since failing slightly earlier has a chance of speeding development activities.

Reply:
This check only makes sense for the rosetta tests, I don't think it should fail all other tests.

Reply:
Just like any sane test framework you can run tests individually which I do locally.

I really don't see the point of such check, it is not like the tests validate other dependencies either and if rosetta test fails it is likely easy enough to figure